app/database/operator/server_data_operator/handlers/post.ts

```diff
@@ -55,6 +54,56 @@ export interface PostHandlerMix {
     handleReceivedPostsInThread: (postsMap: Record<string, Post[]>, prepareRecordsOnly?: boolean) => Promise<PostsInThreadModel[]>;
 }
 
+const mergePostInChannelChunks = async (newChunk: PostsInChannelModel, existingChunks: PostsInChannelModel[]) => {
+    let newChunkUpdated = false;
+    const result: PostsInChannelModel[] = [];
+    for (const chunk of existingChunks) {
+        // Exit early if there is no possibility of any other intersection
+        if (chunk.latest < newChunk.earliest) {
+            break;
+        }
+
+        // omit the current chunk
+        if (chunk.id === newChunk.id) {
+            continue;
+        }
+
+        // delete contained chunks
+        if (newChunk.earliest <= chunk.earliest && newChunk.latest >= chunk.latest) {
+            result.push(chunk.prepareDestroyPermanently());
+            continue;
+        }
+
+        // fuse with any other chunk it intersect with
+        if (
+            (newChunk.earliest <= chunk.earliest && newChunk.latest >= chunk.earliest) ||
```

Comment:
this line looks exactly the same as 72, so it seems this does not apply here.

Reply:
Not really. The difference is on the second part. See that we are checking latest with earliest, not latest with latest, and therefore the intersection.